File: app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel


class HistoryListItem(BaseModel):
    id: int
    data_source_name: str
    table_name: str | None
    prompt: str
    llm_provider: str
    llm_model: str
    execution_status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class HistoryDetail(HistoryListItem):
    generated_code: str | None
    error_message: str | None
    has_image: bool
    duration_ms: int

File: frontend/src/types/history.ts
import type { ExecutionStatus } from "./visualization";

export interface HistoryListItem {
  id: number;
  data_source_name: string;
  table_name: string | null;
  prompt: string;
  llm_provider: string;
  llm_model: string;
  execution_status: ExecutionStatus;
  created_at: string;
}

export interface HistoryDetail extends HistoryListItem {
  generated_code: string | null;
  error_message: string | null;
  has_image: boolean;
  duration_ms: number;
}
